code/modules/reagents/chemistry/reagents/medicine.dm

```diff
@@ -424,13 +424,18 @@
 	addiction_threshold = 5
 	harmless = FALSE
 	taste_description = "health"
+	var/heals_robots = FALSE
 
 /datum/reagent/medicine/omnizine/on_mob_life(mob/living/M)
 	var/update_flags = STATUS_UPDATE_NONE
-	update_flags |= M.adjustToxLoss(-1*REAGENTS_EFFECT_MULTIPLIER, FALSE)
-	update_flags |= M.adjustOxyLoss(-1*REAGENTS_EFFECT_MULTIPLIER, FALSE)
-	update_flags |= M.adjustBruteLoss(-2*REAGENTS_EFFECT_MULTIPLIER, FALSE)
-	update_flags |= M.adjustFireLoss(-2*REAGENTS_EFFECT_MULTIPLIER, FALSE)
+	var/mob/living/carbon/human/ipc = M
+	update_flags |= M.adjustToxLoss(-1 * REAGENTS_EFFECT_MULTIPLIER, FALSE)
+	update_flags |= M.adjustOxyLoss(-1 * REAGENTS_EFFECT_MULTIPLIER, FALSE)
+	update_flags |= M.adjustBruteLoss(-2 * REAGENTS_EFFECT_MULTIPLIER, FALSE)
+	update_flags |= M.adjustFireLoss(-2 * REAGENTS_EFFECT_MULTIPLIER, FALSE)
+	if(heals_robots)
+		update_flags |= ipc.adjustBruteLoss(-2 * REAGENTS_EFFECT_MULTIPLIER, robotic = TRUE)	// This allows IPCs to heal if heals_robots is set to true.
+		update_flags |= ipc.adjustFireLoss(-2 * REAGENTS_EFFECT_MULTIPLIER, robotic = TRUE)		// Because you cannot heal robotic limbs unless you specifically use the human version of adjustXloss. Pain.
```

Comment:
```suggestion
		update_flags |= ipc.adjustBruteLoss(-2 * REAGENTS_EFFECT_MULTIPLIER, robotic = TRUE)
		update_flags |= ipc.adjustFireLoss(-2 * REAGENTS_EFFECT_MULTIPLIER, robotic = TRUE)
```
Also, these should automatically be called on the highest order of the mob - so the typecast shouldn't be necessary?